packages/coil-extension/src/content/services/RunContentHandler.ts

```diff
@@ -24,22 +29,40 @@ export class RunContentHandler {
     private contentRuntime: ContentRuntime,
     @inject(tokens.Window)
     private window: Window,
+    @inject(tokens.Document)
+    private document: Document,
     private frames: Frames,
     private client: GraphQlClient
   ) {}
 
-  async adaptedPageDetails(url: string) {
-    debug('fetching payment pointer for this page', url)
+  async adaptedPageDetails(url: string, site: string) {
+    debug('fetching payment pointer for this page', url, site)
+
+    const variables = { url }
+
+    if (site === 'youtube') {
+      // Wait for page to load and the element containing the channelId is
+      // available
+
+      await this.windowLoaded()
+      const channelUrl = await this.pollForYouTubeChannelId({
```

Comment:
@njlie 
I pushed a commit while I was testing this out. Seems sometimes the element isn't available until a bit after load.